new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioCrossDomainTransaction.java

```diff
@@ -0,0 +1,426 @@
+    // file with the namespaces. for cross domain transaction to work, we need to have the externalDNSName
+    // set in the config file. Cannot set this after the domain is up since a server restart is
+    // required for this to take effect. So, copying the property file to RESULT_ROOT and updating the
+    // property file
+    updatePropertyFile();
+
+    // Label the domain/operator namespace with istio-injection=enabled
+    Map<String, String> labelMap = new HashMap();
+    labelMap.put("istio-injection", "enabled");
+
+    assertDoesNotThrow(() -> addLabelsToNamespace(domain1Namespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(domain2Namespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domain1Namespace, domain2Namespace);
+
+  }
+
+  private static void updatePropertyFile() {
+    //create a temporary directory to copy and update the properties file
+    Path target = Paths.get(PROPS_TEMP_DIR);
+    Path source1 = Paths.get(MODEL_DIR, WDT_MODEL_DOMAIN1_PROPS);
+    Path source2 = Paths.get(MODEL_DIR, WDT_MODEL_DOMAIN2_PROPS);
+    logger.info("Copy the properties file to the above area so that we can add namespace property");
+    assertDoesNotThrow(() -> {
+      Files.createDirectories(target);
+      Files.copy(source1, target.resolve(source1.getFileName()), StandardCopyOption.REPLACE_EXISTING);
+      Files.copy(source2, target.resolve(source2.getFileName()), StandardCopyOption.REPLACE_EXISTING);
+    });
+
+    assertDoesNotThrow(() -> {
+      addNamespaceToPropertyFile(WDT_MODEL_DOMAIN1_PROPS, domain1Namespace);
+      String.format("Failed to update %s with namespace %s",
+            WDT_MODEL_DOMAIN1_PROPS, domain1Namespace);
+    });
+    assertDoesNotThrow(() -> {
+      addNamespaceToPropertyFile(WDT_MODEL_DOMAIN2_PROPS, domain2Namespace);
+      String.format("Failed to update %s with namespace %s",
+            WDT_MODEL_DOMAIN2_PROPS, domain2Namespace);
+    });
+
+  }
+
+  private static void addNamespaceToPropertyFile(String propFileName, String domainNamespace) throws IOException {
+    FileInputStream in = new FileInputStream(PROPS_TEMP_DIR + "/" + propFileName);
+    Properties props = new Properties();
+    props.load(in);
+    in.close();
+
+    FileOutputStream out = new FileOutputStream(PROPS_TEMP_DIR + "/" + propFileName);
+    props.setProperty("NAMESPACE", domainNamespace);
+    props.store(out, null);
+    out.close();
+  }
+
+  /*
+   * This test verifies cross domain transaction is successful. domain in image using wdt is used
+   * to create 2 domains in different namespaces. An app is deployed to both the domains and the servlet
+   * is invoked which starts a transaction that spans both domains.
+   */
+  @Test
+  @DisplayName("Check cross domain transaction with istio works")
+  @Slow
+  @MustNotRunInParallel
```

Comment:
Do we need these 2 tags?

Reply:
I am going to keep it there for now. There is an issue with the istio service port number which might cause a problem if the istio tests are run in parallel. Will remove it later after that issue is resolved..

Reply:
@MustNotRunInParallel  tag may not help. We need find other solution such as adding host name to the istio service to provide host information in http header.